src/routers/groups.ts

```diff
@@ -1,11 +1,23 @@
 import type { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
 import { default as express } from 'express';
 import type * as db from '../db';
-import { getGroups } from '../services/groups';
+import {
+  getGroups,
+  getGroupsByLabelId,
+  getGroupsPerUser,
+  getGroupsPerUserByLabelId,
+} from '../services/groups';
 import { isLoggedIn } from '../middleware/isLoggedIn';
 const router = express.Router();
 
 export function groupsRouter({ dbPool }: { dbPool: PostgresJsDatabase<typeof db> }) {
-  router.get('/', isLoggedIn(dbPool), getGroups(dbPool));
+  router.get('/groups', isLoggedIn(dbPool), getGroups(dbPool));
+  router.get('/groups/:groupLabelId', isLoggedIn(dbPool), getGroupsByLabelId(dbPool));
+  router.get('/groups/:userId', isLoggedIn(dbPool), getGroupsPerUser(dbPool));
+  router.get(
+    '/groups/:userId/:groupLabelId',
```

Comment:
groups is already prepended in the apiRouter, or if it is not then it should be so we do not repeat it here

Reply:
also lets change groupLabelId to groupCategoryId

Reply:
there are a couple issues on the routes here:

1. for express there is no way to differentiate /groups/:groupLabelId and /groups/:userId as they are just wild cards.
2. A hierarchy issue, if we want to do getGroupsPerUser i think it should go to a user router with the structure: `users/:userId/group-categories/:groupCategoryId/groups` and to get all groups for a user i think there is already a route. Also to get groups per group cateogry might require a new router that makes the url look like `/group-categories/:groupCategoryId/groups`